pkg/peertls/tlsopts/options.go

```diff
@@ -64,30 +71,58 @@ func (opts *Options) configure(c Config) (err error) {
 			return Error.Wrap(err)
 		}
 		parseOpts.CAWhitelist = parsed
-		pcvs = append(pcvs, peertls.VerifyCAWhitelist(parsed))
+		opts.VerificationFuncs.ClientAdd(peertls.VerifyCAWhitelist(parsed))
 	}
 
 	if c.Extensions.Revocation {
 		opts.RevDB, err = identity.NewRevDB(c.RevocationDBURL)
 		if err != nil {
 			return err
 		}
-		pcvs = append(pcvs, identity.VerifyUnrevokedChainFunc(opts.RevDB))
+		opts.VerificationFuncs.Add(identity.VerifyUnrevokedChainFunc(opts.RevDB))
 	}
 
 	exts := peertls.ParseExtensions(c.Extensions, parseOpts)
-	pcvs = append(pcvs, exts.VerifyFunc())
+	opts.VerificationFuncs.Add(exts.VerifyFunc())
```

Comment:
We might want to propagate the client/server separation into the extension handling in the future but nothing at the moment requires this.